### ISSUE TITLE
shrink is given a size-indexed function, not data

### ISSUE THREAD

**Issue by athanclark** [jsverify/jsverify#157]
Hi, sorry for posting this here, but it seems to be the only support channel for this project. I'm currently struggling with this issue, and not exactly sure what I'm doing wrong.

I have a data type `Foo` that I'm trying to make an Arbitrary instance for (I'm very familiar with Haskell's QuickCheck. Thank you for making this library, by the way! I have a feeling it will save me a lot of javascript heart ache!). My arbitrary instance looks something like this:

```javascript
myFooArb = {
  generator: jsc.generator.bless(fooGen)
  show: fooShow
  shrink: jsc.shrink.bless(fooShrink)
}
```

Now, here's where I'm going fuzzy:

```javascript
const fooShrink = function(m) {
  console.log("Checking input: " + m); // this should be a `Foo`, right?
  return [new Foo(m.property1, m.property2)] // where property1 and property2 could be constants
}
```

When I try to run the tests, this is what I see being logged:

```
Checking input: function (size) {
    var values = generators.map(function (gen) {
      return gen(size);
    });

    return f.apply(undefined, values);
  }
```

This is where I'm going dizzy. `Foo` itself isn't polymorphic in any way, and my `shrink` function isn't deferring the shrinking behavior to subcomponents - just a single possible shrunken value. In the docs (and in Haskell, too), the shrinking function _should_ consume a `Foo` and return a list/array of `[Foo]`. However, this example shows that it's actually receiving some kind of higher-order generator. How should I be using this `m` parameter? Should I be applying a different combinator to `fooShrink` before setting it as my `{shrink : fooShrink}`?

**Comment by phadej** [comment]
`shrink` is given whatever `generator` have produced. It might bee that `fooGen` is somehow broken. I'd verify it works. For example making some obviously correct property (and maybe omitting shrink, so *jsverify* doesn't try to shrink values at all).

**Comment by athanclark** [comment]
You're completely right, sorry about this. I wasn't returning pure data in the result of my generator. In the docs, you use `... -> generator a` fairly often; I thought that was supposed to be an enriched data structure or something that would be immediately returned by `jsc.generator.foo()`. Sorry again, and thank you!!
